app/js/view/upload_view.js

```diff
@@ -0,0 +1,212 @@
+import { View } from 'components/fxos-mvc/dist/mvc';
+import 'components/gaia-header/dist/gaia-header';
+import 'components/gaia-list/gaia-list';
+import 'gaia-icons';
+import 'components/gaia-text-input/gaia-text-input';
+import 'components/gaia-text-input/gaia-text-input-multiline';
+import { IconHelper } from 'js/lib/helpers';
+import ListModel from 'js/model/list_model';
+
+var UPLOAD_URL = 'http://henretty.us/upload';
+
+export default class UploadView extends View {
+  constructor() {
+    this.el = document.createElement('div');
+    this.el.id = 'uploads';
+    this.el.classList.add('popup');
+    this.currentApp = null;
+
+    this.uploadBlacklist = [
+      'app://ee50b34d-c8ce-b941-8020-5bc4693c770c/manifest.webapp',//FxStumbler
+      'https://webmaker-app.mofostaging.net/manifest.webapp',
+      'app://a9c7eb67-c2c8-8845-b150-5dd0a850200c/manifest.webapp',//Hackerplace
+      'app://7a6c01c5-32fd-d041-9d0c-a70bb6c7c752/manifest.webapp',
+      'app://customizer.gaiamobile.org/manifest.webapp',
+      'app://default_theme.gaiamobile.org/manifest.webapp'
+    ];
+    // Add current directory apps to upload blacklist.
+    var directoryApps = (new ListModel()).getAppList();
+    for (var manifestURL in directoryApps) {
+      this.uploadBlacklist.push(manifestURL);
+    }
+  }
+
+  render() {
+    super();
+    this.closeButton = this.$('.close');
+    this.cancelButton = this.$('#upload-cancel');
+    this.submitButton = this.$('#upload-submit');
+    this.list = this.$('#upload-list');
+    this.displayName = this.$('#display-name');
+    this.displayIcon = this.$('#display-icon');
+    this.nameInput = this.$('#upload-name');
+    this.descriptionInput = this.$('#upload-description');
+    this.alertDialog = document.body.querySelector('#alert-dialog');
+
+    this.closeButton.addEventListener('click', this.hide.bind(this));
+    this.cancelButton.addEventListener('click', this.hideForm.bind(this));
+    this.submitButton.addEventListener('click', this.upload.bind(this));
+    this.createList();
+  }
+
+  showAlertDialog(msg) {
+    this.alertDialog.textContent = msg;
+    this.alertDialog.open();
+  }
+
+  isEligible(app) {
+    if (this.uploadBlacklist.indexOf(app.manifestURL) !== -1) {
+      return false;
+    }
+    if (app.manifest.role === 'theme' || app.manifest.role === 'addon') {
```

Comment:
I believe that this would match the Customizer add-on, which we don't want to allow uploads of.